### PR TITLE
chore: migrate from kuttl to chainsaw (linodeobjectstoragebucket-controller)

### PR DESCRIPTION
**Pull request by eddycharly**
Migrate from kuttl to chainsaw.

### CODE REVIEW

File: e2e/linodeobjectstoragebucket-controller/minimal-linodeobjectstoragebucket/chainsaw-test.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          set -e
+          curl -s \
+            -H "Authorization: Bearer $LINODE_TOKEN" \
+            -H "Content-Type: application/json" \
+            "https://$TARGET_API/$TARGET_API_VERSION/$URI"
+        # TODO check equivalent to grep linodeobjectstoragebucket-sample.us-sea-1.linodeobjects.com
+        # check:
+        #   ($error): ~
+        #   (json_parse($stdout)):
+        #     results: 1
```
Comment (eddycharly) [inline e2e/linodeobjectstoragebucket-controller/minimal-linodeobjectstoragebucket/chainsaw-test.yaml:53]: This check needs to be adapted, please let me know what needs to be checked in the result payload.

File: e2e/linodeobjectstoragebucket-controller/minimal-linodeobjectstoragebucket/chainsaw-test.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - script:
+        content: |-
+          secret=$(kubectl -n $NAMESPACE get secret linodeobjectstoragebucket-sample-access-keys -oyaml)
+          echo $secret | grep read_write
+          echo $secret | grep read_only
```
Comment (eddycharly) [inline e2e/linodeobjectstoragebucket-controller/minimal-linodeobjectstoragebucket/chainsaw-test.yaml:58]: What is this supposed to do ?

**Comment by eddycharly** [comment]
cc @eljohnson92 

i asked a couple of questions to terminate this one ;)

**Comment by eljohnson92** [comment]
@eddycharly I'm taking a look at these tests and working to get them running locally, I'll have a diff for you shortly. is there a way with `chainsaw` to inject a prefix into resources to make then unique per run/environment? I was thinking of using a branch name in the `metadata.name` or something like that.  object store buckets have to be globally unique per region so we are running into some issues with our test that would be great to be able to solve with kuttl

**Comment by eddycharly** [comment]
> is there a way with chainsaw to inject a prefix into resources to make then unique per run/environment?

yes, of course

**Comment by eddycharly** [comment]
@eljohnson92 bump :)

**Comment by eddycharly** [comment]
@eljohnson92 shall i close this one ?

**Comment by eljohnson92** [comment]
Yes, I think it's good to close

**Comment by eddycharly** [comment]
Closing, superseded by #181

**Pull request closed without merging**